linear_algebra/gram_schmidt.cpp

```diff
@@ -0,0 +1,251 @@
+ *
+ * @details
+ * Takes the input of Linearly Independent Vectors,
+ * returns vectors orthogonal to each other.
+ * 
+ * ### Algorithm
+ * Take the first vector of given LI vectors as first vector of Orthogonal vectors.
+ * Take projection of second input vector on the first vector of Orthogonal vector 
+ * and subtract it from the 2nd LI vector.
+ * Take projection of third vector on the second vector of Othogonal vectors and subtract it from the 3rd LI vector.
+ * Keep repeating hte above process until all the vectors in the given input array are exhausted. 
+ *
+ * For Example:
+ * In R2, 
+ * Input LI Vectors={(3,1),(2,2)}
+ * then Orthogonal Vectors= {(3, 1),(-0.4, 1.2)}
+ *
+ *  Have defined maximum dimension of vectors to be 10 and number of vectors
+ *  taken is 20.
+ *  Please do not give linearly dependent vectors
+ *
+ *
+ * @author [Akanksha Gupta](https://github.com/Akanksha-Gupta920)
+ */
+
+#include <iostream>    /// for io operations
+#include <cassert>     /// for assert
+#include <cmath>       /// for fabs
+#include <array>       /// for std::array
+
+/**
+ * @namespace linear_algebra
+ * @brief Linear Algebra algorithms
+ */
+namespace linear_algebra {
+/**
+ * @namespace gram_schmidt
+ * @brief Functions for [Gram Schmidt Orthogonalisation Process] (https://en.wikipedia.org/wiki/Gram%E2%80%93Schmidt_process)
+ */
+namespace gram_schmidt {
+/**
+ * Dot product function.
+ * Takes 2 vectors along with their dimension as input and returns the dot product.
+ * @param x vector 1
+ * @param y vector 2
+ * @param c dimension of the vectors
+ * 
+ * @returns sum
+ */
+double dot_product(const std::array<double, 10>& x, const std::array<double, 10>& y, const int& c) {
+  double sum = 0;
+  for (int i = 0; i < c; i++) {
+    sum += x[i] * y[i];
+  }
+  return sum;
+}
+
+/**
+ * Projection Function
+ * Takes input of 2 vectors along with their dimension and evaluates their projection in temp 
+ *
+ * @param x Vector 1
+ * @param y Vector 2
+ * @param c dimension of each vector
+ *
+ * @returns factor
+ */
+double projection(const std::array<double, 10>& x,const std::array<double, 10>& y,
+                 const int& c) {
+  double dot = dot_product(x, y, c); ///The dot product of two vectors is taken
+  double anorm = dot_product(y, y, c); ///The norm of the second vector is taken.
+  double factor = dot / anorm; ///multiply that factor with every element in a 3rd vector, whose initial values are same as the 2nd vector.
+  return factor;
+}
+
+/**
+ * Function to print the orthogonalised vector
+ *
+ * @param r number of vectors
+ * @param c dimenaion of vectors
+ * @param B stores orthogonalised vectors
+ *
+ * @returns void
+ */
+void display(const int& r,const int& c,const std::array<std::array<double, 10>, 20>& B) {
+  for (int i = 0; i < r; i++) {
+    std::cout << "Vector " << i + 1 << ": ";
+    for (int j = 0; j < c; j++) {
+      std::cout << B[i][j] << " ";
+    }
+    std::cout << std::endl;
```

Comment:
```suggestion
    std::cout << '\n';
```

Reply:
It does not seems reasonable to flush buffer here, so it's good to not to use `std::endl`;
hope this makes sense